pil-analyzer/src/evaluator.rs

```diff
@@ -470,41 +471,54 @@ impl<'a> Definitions<'a> {
 impl<'a, T: FieldElement> SymbolLookup<'a, T> for Definitions<'a> {
     fn lookup(
         &mut self,
-        name: &str,
+        poly: &PolynomialReference,
         type_args: &Option<Vec<Type>>,
     ) -> Result<Arc<Value<'a, T>>, EvalError> {
-        Self::lookup_with_symbols(self.0, name, type_args, self)
+        let impl_pos = type_args
+            .as_ref()
+            .and_then(|type_args| poly.resolved_impls.get(type_args).as_ref().copied());
+
+        match impl_pos {
+            Some(index) => {
+                let parts = poly.name.split("::").collect::<Vec<_>>();
```

Comment:
If a search for `::` is still needed after #1694 then please store the trait name inside the poly reference. At least I think it should be faster that way.